packages/web/src/components/codeMirrorWrapper.tsx

```diff
@@ -115,8 +115,13 @@ export const CodeMirrorWrapper: React.FC<SourceProps> = ({
       return;
 
     let valueChanged = false;
-    if (codemirror.getValue() !== text) {
-      codemirror.setValue(text);
+    // CodeMirror has a bug that renders cursor poorly on a last line.
```

Comment:
File it upstream and add a link here?

Reply:
I am not sure they'll be able to repro, it might be specific to what we do.